app/javascript/src/controllers/score_controller.js

```diff
@@ -3,6 +3,29 @@ import $ from "jquery"
 
 const SCORES = [0, 1, 2, 3, 4, 5]
 
+/* This controller continuously validates all of the score entries in the
+ * assessment and goal setting page. It checks first that the score and goal
+ * are set to allowed values, and verifies that the score never exceeds the
+ * goal.
+ *
+ * A goals page can be of arbitrary length and has an arbitrary set of fields
```

Comment:
having an arbitrary length/fields doesn't prevent us from using stimulus targets since stimulus will generate an array of targets (we use this feature in the plan controller).

Reply:
There's no inherent reason we can't use stimulus targets. In the context of a validation event, we care about the event's target (score or goal input) and its corresponding pair input. Since the controller spans the entire form, it was simpler to find the pair directly with `currentTarget`'s id and `getElementById` than traverse a list of goal or score targets. 
I think another reasonable strategy would be to instantiate one controller per pair, and have `scoreTarget` and `goalTarget`, but this would require us to re-do the input fields interaction with the submit button.